.github/workflows/ci.yml

```diff
@@ -14,6 +14,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
What happens to `ref` when test is triggered by either _push_ or _schedule_ event?

After all do we really need to use `ref` option? Because doc for the option says
```
When checking out the repository that triggered a workflow,
this defaults to the reference or SHA for that event.
```

Reply:
Thanks for the reminder. For flows where we really need the ref we'll either need to use a different var or restrict it to pull request only. For this job I think it's merely a convenience because it makes the hashes embedded in the names match what you'd see if you build it locally. I'd like to leave it here for now because of that convenience and I don't think it'll break anything. If it does, I'll just jump in with another patch to remove it.

Reply:
For _push_ and _schedule_ events, `github.event.pull_request.head.sha` should be unknown. That's my concern. And sorry, I still don't understand the real need of `ref` option.

Reply:
There is no need AFAICT. I will push a new version without it. It does have an *advantage* in that the version numbers and git hashes that get added to the build exactly match those that you'd get if you build locally, but for this case it's not strictly necessary. Furthermore, if github.event.pull_request.head.sha is unavailable the action falls back to it's default, creating a merge commit, it doesn't break anything. Regardless, I will remove it. But you may see it reappear in other places where it actually does serve a significant purpose.

Reply:
> Furthermore, if github.event.pull_request.head.sha is unavailable the action falls back to it's default, creating a merge commit, it doesn't break anything. Regardless, I will remove it. But you may see it reappear in other places where it actually does serve a significant purpose.

Ok.